### DISCUSSION TITLE
Vector Data

### DISCUSSION THREAD

**Discussion by Queliewz** [omnetpp/omnetpp#1292] [Q&A]
Hello @rhornig @giovanninardini, 

I'm running the SingleCell_Standalone simulation and need to extract throughput, packet loss, and delay metrics from the vector data, as shown in the attached image. I noticed that the throughput data contains two values from the same module (SingleCell_Standalone.ue[0].cellularNic.nrRlc.um) which are "rlcPduThroughputDl:vector and "rlcThroughputDl:vector". The same applies to the delay, packet loss, sent and received packet data. Could you please clarify the difference between these two values, apart from their names?

![RESULT F](https://github.com/user-attachments/assets/68028773-95dd-4701-ac66-860a6c3ae52b)

Btw I'm using OMNeT++ 6.0.1, Inet 4.5.4 and Simu5G 1.2.2

**Comment by giovanninardini** [comment]
Hi,

_rlcThroughput_ measures the throughput between the RLC and PDCP layer (received RLC SDUs), wherease _rlcPduThroughput_ measures the throughput between the MAC and the RLC layer (received RLC PDUs).
Note that RLC PDUs are obtained by segmenting/concatenating RLC SDUs at the transmitting node, and include the RLC header as well.

**Reply by Queliewz** [reply]
Oh, I see. Based on the result, the throughput value is 2,697.688823 Bps, which is in Bytes per second (equivalent to 0.02158 Mbps). I’m not sure if I can assume the value should be in Gigabits or Megabits, but it doesn’t make sense for 5G to have a throughput of only 0.02158 Mbps, right? I want to know which unit prefix the data represents.

**Reply by giovanninardini** [reply]
The unit is Bps. 
The throughput depends on a lot of factors. I see that you are probably simulating VoIP traffic: in this case, a VoIP source has quite low data rate (i.e., it sends 40 bytes every 20 ms - not counting silence periods) and of course the throughput cannot be higher than that.

**Reply by Queliewz** [reply]
Yes, I'm simulating for the VoIP. From your explanation, what I understand is that the OMNeT++ simulation software only demonstrates the basic throughput for VoIP, not the actual speed that would be expected on a 5G network. Am I correct?

**Reply by giovanninardini** [reply]
Simu5G can simulate any kind of data traffic, you just need to implement the desired application module. 
The simulator measures the throughput of the specific network scenario you are simulating. In this case you obtain low throughput because VoIP has low data rate, and of course you cannot obtain a throughput higher than the input data rate.
